hydra/controllers/client.py

```diff
@@ -816,3 +817,74 @@ def disable_metrics(self):
 
         self.restart_service()
         self.app.log.info(f'Successfully disabled all metrics reporting.')
+
+    @ex(
+        arguments=[
+            (
+                ['-n', '--name'],
+                {
+                    'help': 'name of network to join',
+                    'action': 'store',
+                    'dest': 'name'
+                }
+            ),
+            (
+                ['-v', '--version'],
+                {
+                    'help': 'version of binary',
+                    'action': 'store',
+                    'dest': 'version'
+                }
+            ),
+            (
+                ['-D', '--destroy'],
+                {
+                    'help': 'destroy existing directory',
+                    'action': 'store_true',
+                    'dest': 'destroy'
+                }
+            ),
+        ]
+    )
+    def upgrade_binary(self, version=None, destroy=False):
+        version = self.app.pargs.version
+
+        p = Prompt(f'Upgrading to ShipChain binary version: {version}. Please confirm (y/n):')
+        if not p.input.lower().startswith('y'):
+            return
+
+        name = self.app.utils.env_or_arg(
+            'name', 'HYDRA_NETWORK', or_path='.hydra_network', required=True)
+
+        destination = self.app.utils.path(name)
+
+        os.chdir(destination)
+
+        self.app.utils.download_release_file('./shipchain-temp', 'shipchain', version)
+
+        os.chmod('./shipchain-temp', os.stat('./shipchain-temp').st_mode | stat.S_IEXEC)
+
+        service_name = f'{name}.service'
+        systemd_service = f'/etc/systemd/system/{service_name}'
+
+        if os.path.exists('./shipchain'):
+            if not self.app.pargs.destroy:
+                self.app.log.info(f'Binary file exists, destroy and replace with -D')
+                return
+            os.remove('./shipchain')
```

Comment:
Probably not a good idea to try to remove the binary before the service is stopped - I'm not sure what would happen here.

Reply:
Yep, not sure why I left that there. Removed.